fix(cli): Expo Go compat deps

frankcalise opened the pull request:
## Describe your PR

- Updates some dependencies to the latest and greatest
- For the ones Expo Go complains aren't compatible, revert those at the time of `package.json` modification when no prebuild is needed

markrickert reviewed (approved):
@frankcalise I added tests to both the expo go and prebuild tests to ensure that the `findAndUpdateDependencyVersions()` function works properly. 

I also moved the expo go configuration and that function to it's own `tools` file so that i could grab the config for easier testing (and so we wouldn't have to update the tests every time we update the expo go config... the tests should just automatically work)

Please look at my commit 7bd6908 to see what was done there.

frankcalise commented:
> @frankcalise I added tests to both the expo go and prebuild tests to ensure that the `findAndUpdateDependencyVersions()` function works properly.
> 
> I also moved the expo go configuration and that function to it's own `tools` file so that i could grab the config for easier testing (and so we wouldn't have to update the tests every time we update the expo go config... the tests should just automatically work)
> 
> Please look at my commit [7bd6908](https://github.com/infinitered/ignite/commit/7bd69089b3729ac5fa2d06a7e54c413097ec59a4) to see what was done there.

Nicely done! Appreciate those updates 🙏
